### PR TITLE
Json parser as a hard copy

### PR DESCRIPTION
**Pull request by wu-hui**
Introduce the json lib as a third party dep by copying the lib into our repo. The reason is there is no official SPM and cocoa pods build spec for the lib, and the owner does not want to maintain those. There is a cocoa pods spec that does not work for us, the the owner of that pod is not maintaining as well. This makes a hard copy the most hassle free approach.

### CODE REVIEW

File: Firestore/core/src/remote/grpc_root_certificates_generated.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #ifndef FIRESTORE_CORE_SRC_REMOTE_GRPC_ROOT_CERTIFICATES_GENERATED_H_
 #define FIRESTORE_CORE_SRC_REMOTE_GRPC_ROOT_CERTIFICATES_GENERATED_H_
 
-#include <stddef.h>
+#include <cstdlib>
```
Comment (var-const) [inline Firestore/core/src/remote/grpc_root_certificates_generated.h:6]: Please revert -- this overwrites the change from #7563.

**Review by var-const — CHANGES_REQUESTED** [review]
* Can you please provide a PR description to explain why this is necessary?
* Have you synced with Paul @paulb777 regarding this approach? In particular, I wonder if there's any legal clearance we need to do to add third-party sources to our repo?

**Comment by paulb777** [comment]
@wu-hui and I discussed this approach this morning and agreed the complexity of managing the copy would be substantially less than managing a CocoaPods podspec and a Swift Package - and it's fine to add MIT licensed code as long as it goes to a third_party directory and is acknowledged.

On thing to consider would be if it's necessary to do symbol renaming to avoid symbol collisions in case Firestore clients could have their own copy of `json.hpp`.

A copy of the LICENSE should appended to https://github.com/firebase/firebase-ios-sdk/blob/master/CoreOnly/NOTICES

**Comment by paulb777** [comment]
On 2nd thought, my C++ is relatively weak, but I don't think a pure header could cause symbol collisions.

File: Firestore/core/src/remote/grpc_root_certificates_generated.h (hunk 1 above)
Comment (wu-hui) [reply Firestore/core/src/remote/grpc_root_certificates_generated.h]: Done.

**Comment by wu-hui** [comment]
> On 2nd thought, my C++ is relatively weak, but I don't think a pure header could cause symbol collisions.

Yeah, I think we should be fine here.

**Comment by var-const** [comment]
Thanks, Paul! I also don't think a header-only library should cause symbol collisions.

**Review by var-const — APPROVED** [review]

**Comment by var-const** [comment]
@wu-hui Andy, can you add some of the rationale for this PR to the PR description?

**Pull request merged**